lib/android/app/src/main/java/com/reactnativenavigation/react/NavigationModule.java

```diff
@@ -279,4 +260,37 @@ public void onCatalystInstanceDestroy() {
         }
         super.onCatalystInstanceDestroy();
     }
+
+    private LifecycleEventListenerAdapter listenerAdapter = new LifecycleEventListenerAdapter() {
+        @Override
+        public void onHostPause() {
+            super.onHostPause();
+            UiUtils.runOnMainThread(() -> {
+                try {
+                    navigator().onHostPause();
+                } catch (Exception e) {
+
+                }
+            });
+        }
+
+        @Override
+        public void onHostDestroy() {
+            super.onHostDestroy();
+            getReactApplicationContext().removeLifecycleEventListener(listenerAdapter);
+        }
+
+        @Override
+        public void onHostResume() {
+            eventEmitter = new EventEmitter(getReactApplicationContext());
+            navigator().setEventEmitter(eventEmitter);
+            layoutFactory.init(
+                    activity(),
+                    eventEmitter,
+                    navigator().getChildRegistry(),
+                    ((NavigationApplication) activity().getApplication()).getExternalComponents()
+            );
+            UiUtils.runOnMainThread(() -> navigator().onHostResume());
```

Comment:
we are calling `onHostResume` again inside `public void onHostResume` wont this be circular? @rahulfancode